src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -174,6 +174,51 @@ void QgsAppDirectoryItemGuiProvider::populateContextMenu( QgsDataItem *item, QMe
   } );
   menu->addAction( hideAction );
 
+  QMenu *hiddenMenu = new QMenu( tr( "Hidden items" ), menu );
+  int count = 0;
+  const QStringList hiddenPathList = settings.value( QStringLiteral( "/browser/hiddenPaths" ) ).toStringList();
+  for ( const QString &path : hiddenPathList )
+  {
+    QAction *action = new QAction( path, hiddenMenu );
+    connect( action, &QAction::triggered, this, [ = ]
+    {
+      QString path = qobject_cast<QAction *>( sender() )->text();
```

Comment:
```suggestion
```

(Path is already captured in the lambda)

Reply:
Done.